File: pkg/java/src/main/java/dev/openfga/language/errors/ModelValidationSingleError.java
package dev.openfga.language.errors;

public class ModelValidationSingleError extends ParsingError {

    private ValidationMetadata metadata;

    // Needed for Jackson deserialization
    public ModelValidationSingleError() {}

    public ModelValidationSingleError(ErrorProperties properties, ValidationMetadata metadata) {
        super("validation", properties);
        this.metadata = metadata;
    }

    public ValidationMetadata getMetadata() {
        return metadata;
    }

    public void setMetadata(ValidationMetadata metadata) {
        this.metadata = metadata;
    }
}

File: pkg/csharp/src/OpenFga.Language/Validation/ModelValidationSingleError.cs
using System.Text.Json.Serialization;
using OpenFga.Language.Errors;

namespace OpenFga.Language.Validation;

public class ModelValidationSingleError : ParsingError
{
    [JsonPropertyName("metadata")]
    public ValidationMetadata Metadata { get; set; } = new();

    public ModelValidationSingleError() { }

    public ModelValidationSingleError(ErrorProperties properties, ValidationMetadata metadata) 
        : base("validation", properties)
    {
        Metadata = metadata;
    }
}
